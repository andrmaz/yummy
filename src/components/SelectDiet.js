import React from 'react';
import PropTypes from 'prop-types';
import { Controller } from 'react-hook-form';
import { Flex, FormLabel, Select} from '@chakra-ui/react';

const SelectDiet = ({ control }) => {
    return (
        <Controller
            name="diet"
            data-testid='c'
            control={control}
            defaultValue='balanced'
            render={({ onChange, value }) => 
            <Flex direction='column'>
                <FormLabel textAlign='center' pt='5px' >Diet</FormLabel>
                <Select 
                    id='diet'
                    onChange={onChange} 
                    value={value} 
                    name='diet'
                    size='xs' 
                    w='200px'
                    m='5px'
                    variant="filled"
                    focusBorderColor='blue.500'
                >
                    <option label='balanced' value='balanced' />
                    <option label='high-protein' value='high-protein' />
                    <option label='high-fiber' value='high-fiber' />
                    <option label='low-fat' value='low-fat' />
                    <option label='low-carb' value='low-carb' />
                </Select>
            </Flex>
            }
        />
    );
};

SelectDiet.propTypes = {
    control: PropTypes.object,
};

export default SelectDiet;
