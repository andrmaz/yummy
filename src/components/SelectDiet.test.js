import React from 'react';
import { shallow } from 'enzyme';
import SelectDiet from './SelectDiet';

let wrapper;

describe('SelectDiet Component', () => {
    beforeEach(() => {
        expect.hasAssertions();
    });
    it('should render Component correctly', () => {
        wrapper = shallow(<SelectDiet />);
        expect(wrapper).toMatchSnapshot();
    });
});