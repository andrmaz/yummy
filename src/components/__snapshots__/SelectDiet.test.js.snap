// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDiet Component should render Component correctly 1`] = `
<Controller
  data-testid="c"
  defaultValue="balanced"
  name="diet"
  render={[Function]}
/>
`;
